middleware/buffer/source/string.cpp

```diff
@@ -129,14 +129,22 @@ namespace casual
                {
                   return CASUAL_STRING_OUT_OF_BOUNDS;
                }
+               catch( const std::invalid_argument&)
+               {
+                  return CASUAL_STRING_INVALID_ARGUMENT;
+               }
                catch( const std::bad_alloc&)
                {
                   return CASUAL_STRING_OUT_OF_MEMORY;
                }
                catch( ...)
                {
-                  if( common::exception::capture().code() == common::code::xatmi::argument)
-                     return CASUAL_STRING_INVALID_HANDLE;
+                  const auto error = common::exception::capture();
+
+                  if( error.code() == common::code::xatmi::argument)
+                     return CASUAL_STRING_INVALID_HANDLE; 
+
+                  common::stream::write( std::cerr, "error: ", error, '\n');
```

Comment:
Same here...

Reply:
Shall we log it into casual-log instead and if so, what category ?

Reply:
We've got the `common::log::error( <code>, <info>...)`. Logs to error category.  But what _code_ should you use? Check if you think you find something sufficient in `code::casual::???`. If not, maybe a `code::buffer` should be created?

Reply:
If you create one, I think it should be local to _buffer_. Should it not?

Reply:
Or, of course, you could add _codes_ in `code::casual` if you want.